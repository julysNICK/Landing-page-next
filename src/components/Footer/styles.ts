import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../Textcomponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
