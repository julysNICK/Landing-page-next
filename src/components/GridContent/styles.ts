import styled, { css } from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
