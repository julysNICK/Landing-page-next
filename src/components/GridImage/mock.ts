import { GridImageProps } from '.';

export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      altText: 'Teste 1',
      srcImg: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      srcImg: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      srcImg: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      srcImg: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      srcImg: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      srcImg: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
} as GridImageProps;
