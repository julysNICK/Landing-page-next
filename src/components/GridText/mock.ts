import { GridTextProps } from '.';

export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
} as GridTextProps;
