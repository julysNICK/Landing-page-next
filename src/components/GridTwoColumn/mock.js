export default {
  title: 'Grid two columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio magnam culpa eveniet doloribus harum? Ipsam, a necessitatibus? Sequi sunt accusantium quod, animi iure a, aliquid dolor ea vel magni dolore?`,
  srcImg: 'assets/images/javascript.svg',
};
