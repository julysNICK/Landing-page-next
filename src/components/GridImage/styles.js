import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
