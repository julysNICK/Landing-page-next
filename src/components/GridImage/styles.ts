import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
