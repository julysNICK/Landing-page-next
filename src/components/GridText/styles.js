import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
