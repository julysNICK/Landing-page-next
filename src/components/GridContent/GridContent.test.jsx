import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { GridContent } from '.';

import mock from './mock';

describe('<GridContent />', () => {
  it('should render grid content', () => {
    const { container } = renderTheme(<GridContent {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render grid content', () => {
    const { container } = renderTheme(
      <GridContent {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
