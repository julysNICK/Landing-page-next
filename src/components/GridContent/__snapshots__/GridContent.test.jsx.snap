// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> should render grid content 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          O título que eu quero
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              <strong>
                Pellentesque habitant morbi tristique
              </strong>
               senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. 
              <em>
                Aenean ultricies mi vitae est.
              </em>
               Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, 
              <code>
                commodo vitae
              </code>
              , ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. 
              <a
                href="#"
              >
                Donec non enim
              </a>
               in turpis pulvinar facilisis. Ut felis.
            </p>
            
  
            <h2>
              Header Level 2
            </h2>
            
  
            <ol>
              
     
              <li>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
              </li>
              
     
              <li>
                Aliquam tincidunt mauris eu risus.
              </li>
              
  
            </ol>
            
  
            <blockquote>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus magna. Cras in mi at felis aliquet congue. Ut a est eget ligula molestie gravida. Curabitur massa. Donec eleifend, libero at sagittis mollis, tellus est malesuada tellus, at luctus turpis elit sit amet quam. Vivamus pretium ornare est.
              </p>
            </blockquote>
            
  
            <h3>
              Header Level 3
            </h3>
            
  
            <ul>
              
     
              <li>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
              </li>
              
     
              <li>
                Aliquam tincidunt mauris eu risus.
              </li>
              
  
            </ul>
            
  
            <pre>
              <code>
                
  #header h1 a {
    display: block;
    width: 300px;
    height: 80px;
  }
  
              </code>
            </pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent /> should render grid content 2`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          O título que eu quero
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              <strong>
                Pellentesque habitant morbi tristique
              </strong>
               senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. 
              <em>
                Aenean ultricies mi vitae est.
              </em>
               Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, 
              <code>
                commodo vitae
              </code>
              , ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. 
              <a
                href="#"
              >
                Donec non enim
              </a>
               in turpis pulvinar facilisis. Ut felis.
            </p>
            
  
            <h2>
              Header Level 2
            </h2>
            
  
            <ol>
              
     
              <li>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
              </li>
              
     
              <li>
                Aliquam tincidunt mauris eu risus.
              </li>
              
  
            </ol>
            
  
            <blockquote>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus magna. Cras in mi at felis aliquet congue. Ut a est eget ligula molestie gravida. Curabitur massa. Donec eleifend, libero at sagittis mollis, tellus est malesuada tellus, at luctus turpis elit sit amet quam. Vivamus pretium ornare est.
              </p>
            </blockquote>
            
  
            <h3>
              Header Level 3
            </h3>
            
  
            <ul>
              
     
              <li>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
              </li>
              
     
              <li>
                Aliquam tincidunt mauris eu risus.
              </li>
              
  
            </ul>
            
  
            <pre>
              <code>
                
  #header h1 a {
    display: block;
    width: 300px;
    height: 80px;
  }
  
              </code>
            </pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
